kahuna/public/js/usage-rights/usage-rights-editor.js

```diff
@@ -6,125 +6,112 @@ import './usage-rights-editor.css!';
 
 import '../components/gr-confirm-delete/gr-confirm-delete.js';
 
+
+import Rx from 'rx';
+import '../util/rx';
+
 export var usageRightsEditor = angular.module('kahuna.edits.usageRightsEditor', [
     'monospaced.elastic',
-    'gr.confirmDelete'
+    'gr.confirmDelete',
+    'util.rx'
 ]);
 
 usageRightsEditor.controller(
     'UsageRightsEditorCtrl',
-    ['$q', '$scope', '$window', '$timeout', 'editsService', 'editsApi', 'onValChange',
-    function($q, $scope, $window, $timeout, editsService, editsApi, onValChange) {
+    ['$q', '$scope', 'inject$', 'observe$', 'editsService', 'editsApi',
+    function($q, $scope, inject$, observe$, editsService, editsApi) {
 
     var ctrl = this;
+    const multiCat = { name: 'Multiple categories', value: 'multi-cat', properties: [] };
 
-    ctrl.saving = false;
-    ctrl.saved = false;
-    ctrl.categories = [];
-    ctrl.originalCats = [];
-    ctrl.model = angular.extend({}, ctrl.usageRights.data);
-
-    const multiRights = { name: 'Multiple categories', value: '' };
-    const noRights = { name: 'None', value: '' };
-    const catsWithNoRights = () => [noRights].concat(ctrl.originalCats);
-    const catsWithMultiRights = () => [multiRights].concat(ctrl.originalCats);
-    const setStandardCats = () => ctrl.categories = ctrl.originalCats;
-    const setNoRightsCats = () => {
-        ctrl.categories = catsWithNoRights();
-        ctrl.category = noRights;
-    };
-    const setMultiRightsCats = () => {
-        ctrl.categories = catsWithMultiRights();
-        ctrl.category = multiRights;
-    };
-
+    // @return Stream.<Array.<UsageRights>>
+    const usageRights$ = observe$($scope, () => ctrl.usageRights).startWith([]);
```

Comment:
It does work - I've always seemed to have settled for the function method as it fells less magic.

We seem to have most commnly gone with the function method on `$watch`es, but assuming this is because a `$watch` can't take a string.

I can't find any documentation on performance, so happy to go with either.